packages/flutter_tools/lib/src/custom_devices/custom_device.dart

```diff
@@ -621,6 +624,15 @@ class CustomDevice extends Device {
       return _getOrCreateAppSession(app).logReader;
     }
 
+    if (_config.supportsReadingLogs && _globalLogReaderProcess == null) {
+      // launch the readLogs command
+      return _processUtils.start(_config.readLogsCommand!).then((Process process) {
```

Comment:
This should use `async`/`await` and not `.then(...)`

Reply:
Sounds good. I'll make the change.

Reply:
Done. Sorry it took so long.